lib/systemFlake.nix

```diff
@@ -94,8 +101,16 @@ let
   getChannels = system: self.pkgs.${system};
   getNixpkgs = host: (getChannels host.system).${host.channelName};
 
-  configurationBuilder = hostname: host': (
+  configurationBuilder = reverseDnsFqdn: host': (
     let
+      dnsLabels = reverseList (partitionString "\\." reverseDnsFqdn);
+      hostname = builtins.head dnsLabels;
+      domain = let
+        domainLabels = builtins.tail dnsLabels;
```

Comment:
Makes me wonder if we actually need to reverse it, users could just type already reversed domain.

Reply:
Tho its fine, if you guys prefer it we can definitely leave it as it is

Reply:
The reason to have reverse dns is two fold:

- It translates nicely and dorectly into a folder structure `./com/example/myhost`
- the domain grouping is preserved through alphanumerical sorting. For example in a repl exploring the `nixosConfigurations`

Reply:
> It translates nicely and directly into a folder structure ./com/example/myhost (or any other nested attribute set)

Maybe we should reverse it in digga instead. For fup users it makes more sense for them to define their hosts with literal domain names.
```
hosts."myhost.example.com"
```
is much more obvious and simple to understand for in a fup flake.

> the domain grouping is preserved through alphanumerical sorting. For example in a repl exploring the nixosConfigurations

I don't see how this is exclusive to reverse dns.

Reply:
> I don't see how this is exclusive to reverse dns. [useful alphanumerical sorting]

Semantically useful (grouped by zone):
```
com.example.a
com.example.b
de.example.a
de.example.b
```

Not very useful:
```
a.example.com
a.example.de
b.example.com
b.example.de
```

Reply:
One may argue that it would be more convenient this way due to people straight up typing the machine name is more unique compared to the domain
```
GytisMachine.xyz.example.com
PacmanMachine.xyz.example.com
DavidMachine.xyz.example.com
```

Reply:
I'm still trying to figure out if I'm only upholding a "theoretical" benefit here.

My gut tells me there is something valid to reverse DNS, but sometimes my gut is wrong.

I'll meditate over it. :wink:

One thing that might be bothersome is that the api between `digga` and `fup` might diverge. Although maybe the magic digga importer can already deal with it.

Still meditating.

Reply:
> One thing that might be bothersome is that the api between digga and fup might diverge. Although maybe the magic digga importer can already deal with it.

This is only true for the auto-importer where it makes sense for folder structure to use reverse dns - it would be awkward to have a `com.nix` in a folder.

I do see some historical use of reverse dns in programming, freedesktop and java. It seems to be pretty common in programming, so perhaps we should just stick with it. Those groups seem to have identified advantages with reverse dns notation. But it does make fup user facing api kind of complicated.

Reply:
So I tossed a coin and it said: reverse DNS. So that it'll be. :wink:

Reply:
cool, can you at least add a comment under the function argument to clarify what you mean by reverse dns, like I suggested in the other comment(adapted to reverse dns of course).
Reverse dns is too ambigous and confusing of a name.